pyaedt/modules/SolveSetup.py

```diff
@@ -493,24 +510,60 @@ def add_mesh_link(self, design_name, solution_name, parameters_dict, project_nam
 
         >>> oModule.EditSetup
         """
-        meshlinks = self.props["MeshLink"]
-        meshlinks["ImportMesh"] = True
-        meshlinks["Project"] = project_name
-        meshlinks["Product"] = "ElectronicsDesktop"
-        meshlinks["Design"] = design_name
-        meshlinks["Soln"] = solution_name
-        meshlinks["Params"] = OrderedDict({})
-        for el in parameters_dict:
-            if el in list(self._app.available_variations.nominal_w_values_dict.keys()):
-                meshlinks["Params"][el] = el
+        try:
+            if self.p_app.design_type == "Mechanical":
+                design_type = "ElectronicsDesktop"
             else:
-                meshlinks["Params"][el] = parameters_dict[el]
-        meshlinks["ForceSourceToSolve"] = True
-        meshlinks["PreservePartnerSoln"] = True
-        meshlinks["PathRelativeTo"] = "TargetProject"
-        meshlinks["ApplyMeshOp"] = True
-        self.update()
-        return True
+                design_type = self.p_app.design_type
+            if not design_name or design_name is None:
+                raise ValueError("Provide design name to add mesh link to.")
+            # if self.p_app.solution_type == "SBR+":
+            meshlinks = self.props["MeshLink"]
+            meshlinks["ImportMesh"] = True
+            meshlinks["Product"] = design_type
+            if project_name != "This Project*":
+                if os.path.exists(project_name):
+                    meshlinks["Project"] = project_name
+                    meshlinks["PathRelativeTo"] = "SourceProduct"
+                else:
+                    raise ValueError("Project file path provided does not exist.")
+            else:
+                meshlinks["Project"] = project_name
+                meshlinks["PathRelativeTo"] = "TargetProject"
+                if design_name not in self.p_app.design_list:
+                    raise ValueError("Design does not exist in current project.")
+                meshlinks["Design"] = design_name
+                if solution_name is None:
+                    meshlinks["Soln"] = "{} : LastAdaptive".format(
+                        self.p_app.oproject.GetDesign(design_name).GetChildObject("Analysis").GetChildNames()[0]
+                    )
+                elif (
+                    solution_name.split()[0]
+                    in self.p_app.oproject.GetDesign(design_name).GetChildObject("Analysis").GetChildNames()
+                ):
+                    meshlinks["Soln"] = "{} : LastAdaptive".format(solution_name.split()[0])
+                else:
+                    raise ValueError("Setup does not exist in current design.")
+
+            meshlinks["Params"] = OrderedDict({})
+            if parameters_dict is None:
+                parameters_dict = self.p_app.available_variations.nominal_w_values_dict
+                for el in parameters_dict:
+                    meshlinks["Params"][el] = el
```

Comment:
```suggestion
                    meshlinks["Params"][el] = parameters_dict[el]
```

Reply:
@MaxJPRey  
actually I kept the original version: meshlinks["Params"][el] = el
because when you add mesh link if you don't map variables by name it gives back an error and the link is not created.
Look at this example:
![image](https://user-images.githubusercontent.com/103059376/203574982-f2300f5d-252c-4a6c-bce9-c64453e4c41c.png)
I slightly modified the test though.

Reply:
OK, makes sense. Thanks for the explanation @gmalinve .